docs/reference/transforms/task_context.rst

```diff
@@ -0,0 +1,135 @@
+.. _task_context:
+
+Task Context
+============
+
+The :mod:`taskgraph.transforms.task_context` transform can be used to
+substitute values into any field in a task with data that is not known
+until ``taskgraph`` runs.
+
+This data can be provided in a few ways, as described below.
+
+Usage
+-----
+
+from-parameters
+~~~~~~~~~~~~~~~
+
+Context can be pulled from parameters that are provided to ``taskgraph``.
+
+First, add the transform to the ``transforms`` key in your ``kind.yml`` file:
+
+.. code-block:: yaml
+
+   transforms:
+     - taskgraph.transforms.task_context
+     # ...
+
+Then create a ``task-context`` section in your task definition, e.g:
+
+.. code-block:: yaml
+
+   tasks:
+     build:
+       description: my description {foo}
+       task-context:
+         from-parameters:
+           foo: foo
```

Comment:
Ah so, these aren't necessarily the same as parameters in `parameters.yml`? Can we use a different name in that case? Maybe `from-object`? Or `from-dict`?

Reply:
These are Parameters. The keys here are the names you'll use in the tasks, the values are where to find them in `parameters.yml`. (If you're wondering why we can't just use a list of strings...I guess we could, although it's not as obvious what you use in the `kind` when you're retrieving nested values, eg: https://github.com/mozilla/firefox-translations-training/blob/main/taskcluster/ci/collect-mono-trg/kind.yml#L27.)

Reply:
Ah ok, my mistake. Thanks for clarifying!

I guess you could allow specifying nested values in the string as well and then the name could just include the dots or replace them with `_` or something... But this seems fine as well.